pagefaultgames/pokerogue
# [Refactor] Added `BattlerTagLapseType.AFTER_HIT`

## What are the changes the user will see?
No changes user-side

## Why am I making these changes?
Beak Blast and Shell Trap both implement `BattlerTags` that have some form of effect when the tag holder is hit but currently, each such effect needs to be handled and checked for individually. Adding a new `LapseType` to handle these similar tags all at once should help reduce duplicate code, standardize their implementations, and streamline future development.

## What are the changes from a developer perspective?
- `BattlerTagLapseType.AFTER_HIT` has been added
- The implementations of Beak Blast and Shell Trap have been slightly adjusted to use this lapse type
- Instead of handling each tag individually, `MoveEffectPhase` now lapses all `AFTER_HIT` tags using `lapseTags()`
- Minor utility functions have been added in `pokemon.ts` and `battler-tags.ts`

## How to test the changes?
There should be no change in gameplay, so just proceed as you normally would and watch for any unexpected bugs.
There are also existing tests for Beak Blast and Shell Trap:
`npm run test beak_blast`
`npm run test shell_trap`

## Checklist
